src/app/compressed-air-assessment/system-profile/profile-setup-form/profile-setup-form.component.ts

```diff
@@ -28,19 +28,23 @@ export class ProfileSetupFormComponent implements OnInit {
   halfHourIntervalData: Array<ProfileSummary>;
   quarterHourIntervalData: Array<ProfileSummary>;
   isProfileDataTypeChange: boolean = false;
+  dayTypesWarningMessage: string = 'is valid';
+  compressedAirAssessment: CompressedAirAssessment;
   constructor(private systemProfileService: SystemProfileService, private compressedAirAssessmentService: CompressedAirAssessmentService,
     private performancePointsFormService: PerformancePointsFormService) { }
 
   ngOnInit(): void {
     this.compressedAirAssessmentSub = this.compressedAirAssessmentService.compressedAirAssessment.subscribe(val => {
       this.dayTypes = val.compressedAirDayTypes;
+      this.compressedAirAssessment = val;
       this.setPressureMinAndMax(val.compressorInventoryItems);
       if (val && this.isFormChange == false) {
         this.form = this.systemProfileService.getProfileSetupFormFromObj(val.systemProfile.systemProfileSetup, this.dayTypes);
         this.enableDisableForm();
       } else {
         this.isFormChange = false;
-      }
+      }  
+      this.checkDayTypesForData();
```

Comment:
Should move this up under the if block. I think this is currently called an extra time if the compressedAirAssessment change originates from this form save(); other than that everything's gravy